src/test/codewhisperer/service/telemetry.test.ts

```diff
@@ -0,0 +1,657 @@
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+            await typing(editor, 'H')
+            // assert.strictEqual(editor.document.getText(), 'H')
+
+            const anotherEditor = await openATextEditorWithText('', 'anotherTest.py')
+            await manualTrigger(anotherEditor, client, config)
+            await waitUntilSuggestionSeen()
+            await acceptByTab()
+            // assert.strictEqual(anotherEditor.document.getText(), `Baz${os.EOL}Baz`)
+
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
+                session2UserTriggerEvent(),
+            ])
+        })
+    })
+
+    describe('on editor change, focus change', function () {
+        it('reject: trigger then open another editor', async function () {
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py', tempFolder, { preview: false })
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+
+            await openATextEditorWithText('text in 2nd editor', 'another1.py', tempFolder, {
+                preview: false,
+            })
+            const anotherEditor = await openATextEditorWithText('text in 3rd editor', 'another2.py', tempFolder, {
+                preview: false,
+            })
+
+            assert.strictEqual(vscode.window.activeTextEditor, anotherEditor)
+            // assert.strictEqual(editor.document.getText(), '')
+            await assertTabCount(3)
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
+            ])
+        })
+
+        it('reject: trigger then close editor', async function () {
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py')
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+            await closeActiveEditor()
+            // assert.strictEqual(editor.document.getText(), '')
+
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
+            ])
+        })
+
+        it('reject: onFocusChange', async function () {
+            assertSessionClean()
+            const editor = await openATextEditorWithText('', 'test.py')
+
+            await manualTrigger(editor, client, config)
+            await waitUntilSuggestionSeen()
+            // assert.strictEqual(editor.document.getText(), '')
+
+            await RecommendationHandler.instance.onFocusChange()
+            assertTelemetry('codewhisperer_userTriggerDecision', [
+                session1UserTriggerEvent({ codewhispererSuggestionState: 'Reject' }),
```

Comment:
most of tests are only testing sessionId, requestId, and decision, can we cover more fields from the userTriggerDecision event? not a blocker for this PR though, thanks for adding these!

Reply:
yea that would be incremental progress, we now only add tests for userTriggerDecision which is considered as p0